x/mongo/driver/topology/pool.go

```diff
@@ -524,69 +405,104 @@ func (p *pool) getGenerationForNewConnection(serviceID *primitive.ObjectID) uint
 	return p.generation.addConnection(serviceID)
 }
 
-// removeConnection removes a connection from the pool.
-func (p *pool) removeConnection(c *connection, reason string) error {
-	if c.pool != p {
+// removeConnection removes a connection from the pool and emits a "ConnectionClosed" event.
+func (p *pool) removeConnection(conn *connection, reason string) error {
```

Comment:
The previous implementation didn't have it, but I do think it's a good idea to add since it would cause a panic if that condition ever happened. Will add.

Reply:
Added nil check.